Tests/DatadogTests/Datadog/RUM/RUMVitals/VitalMemoryReaderTest.swift

```diff
@@ -0,0 +1,60 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import XCTest
+@testable import Datadog
+
+internal class VitalMemoryReaderTest: XCTestCase {
+    func testReadMemory() throws {
+        let reader = VitalMemoryReader()
+
+        let result = reader.readVitalData()
+
+        XCTAssertNotNil(result)
+    }
+
+    func testWhenMemoryConsumptionGrows() {
+        // Given
+        let reader = VitalMemoryReader()
+        let threshold = reader.readVitalData()
+        let allocatedSize = 128 * 1_024
+
+        // When
+        let intPointer = UnsafeMutablePointer<Int>.allocate(capacity: allocatedSize)
+        for i in 0..<allocatedSize {
+            (intPointer + i).initialize(to: i)
+        }
+        let measure = reader.readVitalData()
```

Comment:
The way I read from the docs about the `UnsafeMutablePointer` the `capacity` should be the number of memory blocks specific for that type. The way I understand this is that the memory consumed should be `allocatedSize * 32`, isn't this correct ?